framework/src/meshdivisions/SubdomainsDivision.C

```diff
@@ -38,9 +38,7 @@ SubdomainsDivision::initialize()
   if (!blockRestricted())
   {
     // The subdomains may not be contiguously numbered
-    std::set<libMesh::subdomain_id_type> subdomain_ids;
-    _mesh.getMesh().subdomain_ids(subdomain_ids, /*global=*/true);
-
+    const auto subdomain_ids = blockIDs();
```

Comment:
Maybe you would get copy elision here, but you can catch by reference with this one

Reply:
fixed